docs/public-networks/reference/cli/options.md

```diff
@@ -6028,6 +6028,48 @@ instead of headers only, allowing full historical data to be retained. The defau
 
 Setting this option to `false` increases sync time and disk space usage.
 
+### `snapsync-synchronizer-transaction-indexing-enabled`
+
+<Tabs>
+
+<TabItem value="Syntax" label="Syntax" default>
+
+```bash
+--snapsync-synchronizer-transaction-indexing-enabled[=<true|false>]
+```
+
+</TabItem>
+
+<TabItem value="Example" label="Example">
+
+```bash
+--snapsync-synchronizer-transaction-indexing-enabled=true
+```
+
+</TabItem>
+
+<TabItem value="Environment variable" label="Environment variable">
+
+```bash
+BESU_SNAPSYNC_SYNCHRONIZER_TRANSACTION_INDEXING_ENABLED=true
+```
+
+</TabItem>
+
+<TabItem value="Configuration file" label="Configuration file">
+
+```bash
+snapsync-synchronizer-transaction-indexing-enabled=true
+```
+
+</TabItem>
+
+</Tabs>
+
+Enables or disables transaction indexing during initial sync for [snap sync](../../concepts/node-sync.md#snap-synchronization) and [checkpoint sync](../../concepts/node-sync.md#checkpoint-synchronization).
+Enable this option to query historical transactions by hash during the initial sync.
```

Comment:
```suggestion
Enable this option to query historical transactions by hash.

Setting this option to `true` increases sync time and disk space usage.
```

It's possible to query historical transactions during the sync or after sync regardless of this option so think it's clearer not to mention the initial sync especially since most people would run queries after the sync.

Also think we should mention that will increase sync time and disk space if enabled.